tst/EnergyPlus/unit/VariableSpeedCoils.unit.cc

```diff
@@ -2807,4 +2810,165 @@ TEST_F(EnergyPlusFixture, VariableSpeedCoils_Test_CalcTotCap_VSWSHP)
+        "    PLF Curve, 0.85, 0.8333, 0.0, 0.0, 0.3, 0.85, 1.0, Dimensionless, Dimensionless;",
+
+        });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+    // get coil inputs
+    VariableSpeedCoils::GetVarSpeedCoilInput(*state);
+    // Setting predefined tables is needed though
+    OutputReportPredefined::SetPredefinedTables(*state);
+    // Set up some environmental parameters
+    state->dataEnvrn->OutDryBulbTemp = 5.0;
+    state->dataEnvrn->OutHumRat = 0.0009;
+    state->dataEnvrn->OutBaroPress = 99000.0;
+    state->dataEnvrn->WindSpeed = 5.0;
+    state->dataEnvrn->WindDir = 270.0;
+    // set coil parameters
+    int const CyclingScheme = DataHVACGlobals::ContFanCycCoil;
+    int DXCoilNum = 1;
+    int CompOp = 0;
+    int const SpeedCal = 1;
+    Real64 RuntimeFrac = 1.0;
+    Real64 SensLoad = 0.0;
+    Real64 LatentLoad = 0.0;
+    Real64 PartLoadFrac = 0.0;
+    Real64 OnOffAirFlowRatio = 1.0;
+    Real64 SpeedRatio = 0.0;
+    Real64 MaxONOFFCyclesperHour(0.0);
+    Real64 HPTimeConstant(0.0);
+    Real64 FanDelayTime(0.0);
+
+    // run coil init
+    VariableSpeedCoils::InitVarSpeedCoil(
+        *state, DXCoilNum, MaxONOFFCyclesperHour, HPTimeConstant, FanDelayTime, SensLoad, LatentLoad, CyclingScheme, OnOffAirFlowRatio, SpeedRatio, SpeedCal);
+    // set coil inlet condition
+    state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).InletAirDBTemp = 24.0;
+    state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).InletAirHumRat = 0.009;
+    state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).InletAirEnthalpy = Psychrometrics::PsyHFnTdbW(24.0, 0.009);
+    // test 1: compressor is On but PLR = 0
+    CompOp = 1;
+    PartLoadFrac = 0.0;
+    // set coil inlet air flow rate to speed 1
+    state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).AirMassFlowRate =
+        state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).MSRatedAirMassFlowRate(1) * 0.1;
+    state->dataVariableSpeedCoils->LoadSideMassFlowRate = state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).AirMassFlowRate;
+    DataLoopNode::Node(state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).AirInletNodeNum).MassFlowRate =
+        state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).AirMassFlowRate;
+    VariableSpeedCoils::CalcVarSpeedCoilCooling(
+        *state, DXCoilNum, CyclingScheme, RuntimeFrac, SensLoad, LatentLoad, CompOp, PartLoadFrac, OnOffAirFlowRatio, SpeedRatio, SpeedCal);
+    ;
+    // check coil outlet and inlet air conditions match
+    EXPECT_EQ(state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).OutletAirDBTemp, state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).InletAirDBTemp);
+    EXPECT_EQ(state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).OutletAirHumRat, state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).InletAirHumRat);
+    EXPECT_EQ(state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).OutletAirEnthalpy, state->dataVariableSpeedCoils->VarSpeedCoil(DXCoilNum).InletAirEnthalpy);
```

Comment:
This shows the coil outlet node air condition matches that of the inlet air condition when the PLR is zero.